README.md

```diff
@@ -1,2 +1,144 @@
+
+> A [DID](https://w3c-ccg.github.io/did-spec/) (Decentralized Identifier) method driver for the `did-io` library and for standalone use
+
+## Table of Contents
+
+- [Security](#security)
+- [Background](#background)
+- [Supported Drivers](#supported-drivers)
+- [Install](#install)
+- [Usage](#usage)
+- [Contribute](#contribute)
+- [Commercial Support](#commercial-support)
+- [License](#license)
+
+## Security
+
+TBD
+
+## Background
+
+A `did:key` method driver for the [`did-io`](https://github.com/digitalbazaar/did-io)
+client library and for standalone use.
+
+The `did:key` method is used to express public keys in a way that doesn't 
+require a DID Registry of any kind. Its general format is:
+
+```
+did:key:<multibase encoded, multicodec identified, public key>
+```
+
+So, for example, the following DID would be derived from a base-58 encoded 
+ed25519 public key:
+
+```
+did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH
+```
+
+See also (related specs):
+
+* [Decentralized Identifiers (DIDs) - Data Model and Syntaxes](https://w3c-ccg.github.io/did-spec/)
+* [Linked Data Cryptographic Suite Registry](https://w3c-ccg.github.io/ld-cryptosuite-registry/)
+* [Linked Data Proofs](https://w3c-dvcg.github.io/ld-proofs/)
+
+## Install
+
+Requires Node.js 8.3+
+
+To install locally (for development):
+
+```
+git clone https://github.com/digitalbazaar/did-method-key.git
+cd did-method-key
+npm install
+```
+
+## Usage
+
+To generate a new key and get its corresponding `did:key` method DID Document:
+
+```js
+const didKeyDriver = require('did-method-key').driver();
+
+const didDocument = await didKeyDriver.generate(); // Ed25519 key type by default
+
+JSON.stringify(didDocument, null, 2);
+```
+
+To get a DID Document for an existing `did:key` DID:
+
+```js
+const didDocument = await didKeyDriver.get({did: 'did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH'});
+```
+
+Results in:
+
+```json
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+  "publicKey": [
+    {
+      "id": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+      "type": "Ed25519VerificationKey2018",
+      "controller": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH",
+      "publicKeyBase58": "B12NYF8RrR3h41TDCTJojY59usg3mbtbjnFs7Eud1Y6u"
+    }
+  ],
+  "authentication": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "assertionMethod": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "capabilityDelegation": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "capabilityInvocation": [
+    "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+  ],
+  "keyAgreement": [
+    {
+      "id": "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH#zBzoR5sqFgi6q3iFia8JPNfENCpi7RNSTKF7XNXX96SBY4",
```

Comment:
@dlongley Just to double check - is this `keyAgreement.id` in the format that we expect? (Specifically, `did:key:<ed25519 fingerprint>#<curve25519 fingerprint>`)

Reply:
Yes